database/migrations/0001_init_podcasting_20_database.sql

```diff
@@ -0,0 +1,844 @@
+        'mixed', 'podcastL', 'musicL', 'videoL', 'filmL', 'audiobookL', 'newsletterL', 'blogL', 'publisherL', 'courseL'
+    ))
+);
+
+INSERT INTO medium_value (value) VALUES
+    ('publisher'),
+    ('podcast'), ('music'), ('video'), ('film'), ('audiobook'), ('newsletter'), ('blog'), ('course'),
+    ('mixed'), ('podcastL'), ('musicL'), ('videoL'), ('filmL'), ('audiobookL'), ('newsletterL'), ('blogL'), ('publisherL'), ('courseL')
+;
+
+----------** TABLES **----------
+
+--** FEED > FLAG STATUS
+
+-- used internally for identifying and handling spam and other special flag statuses.
+CREATE TABLE feed_flag_status (
+    id SERIAL PRIMARY KEY,
+    status TEXT UNIQUE CHECK (status IN ('none', 'spam', 'takedown', 'other', 'always-allow')),
+    created_at server_time_with_default,
+    updated_at server_time_with_default
+);
+
+CREATE TRIGGER set_updated_at_feed_flag_status
+BEFORE UPDATE ON feed_flag_status
+FOR EACH ROW
+EXECUTE FUNCTION set_updated_at_field();
+
+INSERT INTO feed_flag_status (status) VALUES ('none'), ('spam'), ('takedown'), ('other'), ('always-allow');
+
+--** FEED
+
+-- The top-level table for storing feed data, and internal parsing data.
+CREATE TABLE feed (
+    id SERIAL PRIMARY KEY,
+    url varchar_url UNIQUE NOT NULL,
+
+    -- feed flag
+    feed_flag_status_id INTEGER NOT NULL REFERENCES feed_flag_status(id),
+
+    -- internal
+
+    -- Used to prevent another thread from parsing the same feed.
+    -- Set to current time at beginning of parsing, and NULL at end of parsing. 
+    -- This is to prevent multiple threads from parsing the same feed.
+    -- If is_parsing is over X minutes old, assume last parsing failed and proceed to parse.
+    is_parsing server_time,
+
+    -- 0 will only be parsed when PI API reports an update.
+    -- higher parsing_priority will be parsed more frequently on a schedule.
+    parsing_priority INTEGER DEFAULT 0 CHECK (parsing_priority BETWEEN 0 AND 5),
+
+    -- the run-time environment container id
+    container_id VARCHAR(12),
+
+    created_at server_time_with_default,
+    updated_at server_time_with_default
+);
+
+CREATE TRIGGER set_updated_at_feed
+BEFORE UPDATE ON feed
+FOR EACH ROW
+EXECUTE FUNCTION set_updated_at_field();
+
+CREATE TABLE feed_log (
+    id SERIAL PRIMARY KEY,
+    feed_id INTEGER NOT NULL REFERENCES feed(id) ON DELETE CASCADE,
+    last_http_status INTEGER,
+    last_crawl_time server_time,
+    last_good_http_status_time server_time,
+    last_parse_time server_time,
+    last_update_time server_time,
+    crawl_errors INTEGER DEFAULT 0,
+    parse_errors INTEGER DEFAULT 0
+);
+
+--** CHANNEL
+
+-- <channel>
+CREATE TABLE channel (
+    id SERIAL PRIMARY KEY,
+    id_text short_id_v2 UNIQUE NOT NULL,
+    slug varchar_slug,
+    feed_id INTEGER NOT NULL REFERENCES feed(id) ON DELETE CASCADE,
+    podcast_index_id INTEGER UNIQUE NOT NULL,
+    podcast_guid UUID UNIQUE, -- <podcast:guid>
```

Comment:
Can this be null at any point?

Reply:
Yes it can be null. Not every RSS feed will have a `<podcast:guid>` tag (although it is highly encouraged they do).